server/controllers/api/videos/view.ts

```diff
@@ -43,8 +45,6 @@ async function viewVideo (req: express.Request, res: express.Response) {
   }
 
   await updateUserHistoryIfNeeded(body, video, res)
-
-  return res.status(HttpStatusCode.NO_CONTENT_204).end()
```

Comment:
I don't think we should always return 204 here. Compared to the contact form where redis is used to for rate limit but the email is correctly sent, in this case the view is not correctly handled meaning this endpoint failed so we should return a error to the client.